File: src/main/java/com/lessmarkup/userinterface/nodehandlers/user/ForgotPasswordNodeHandler.java
package com.lessmarkup.userinterface.nodehandlers.user;

import com.google.inject.Inject;
import com.lessmarkup.Constants;
import com.lessmarkup.TextIds;
import com.lessmarkup.framework.helpers.DependencyResolver;
import com.lessmarkup.framework.helpers.LanguageHelper;
import com.lessmarkup.interfaces.cache.DataCache;
import com.lessmarkup.interfaces.structure.ChildHandlerSettings;
import com.lessmarkup.interfaces.structure.NodeAccessType;
import com.lessmarkup.userinterface.model.user.ForgotPasswordModel;
import com.lessmarkup.userinterface.nodehandlers.common.DialogNodeHandler;

import java.util.Objects;

public class ForgotPasswordNodeHandler extends DialogNodeHandler<ForgotPasswordModel> {

    @Inject
    public ForgotPasswordNodeHandler(DataCache dataCache) {
        super(dataCache, ForgotPasswordModel.class);
    }

    @Override
    protected String getApplyCaption() {
        return LanguageHelper.getText(Constants.ModuleTypeMain(), TextIds.RESTORE_PASSWORD);
    }

    @Override
    protected ForgotPasswordModel loadObject() {
        return DependencyResolver.resolve(ForgotPasswordModel.class);
    }

    @Override
    protected String saveObject(ForgotPasswordModel changedObject) {
        changedObject.submit(this, getFullPath());
        return null;
    }

    @Override
    public ChildHandlerSettings getChildHandler(String path) {
        if (path == null) {
            return null;
        }

        String[] parts = path.split("/");

        if (parts.length != 3 || !Objects.equals(parts[0], "ticket")) {
            return null;
        }

        ResetPasswordNodeHandler handler = DependencyResolver.resolve(ResetPasswordNodeHandler.class);
        handler.initialize(null, null, null, null, NodeAccessType.READ);
        handler.initialize(parts[1], parts[2]);

        ChildHandlerSettings ret = new ChildHandlerSettings();
        ret.setHandler(handler);
        ret.setPath(path);
        return ret;
    }
}

File: src/main/scala/com/lessmarkup/userinterface/nodehandlers/user/ForgotPasswordNodeHandler.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.userinterface.nodehandlers.user

import com.google.inject.Inject
import com.lessmarkup.framework.helpers.{DependencyResolver, LanguageHelper}
import com.lessmarkup.framework.nodehandlers.NodeHandlerConfiguration
import com.lessmarkup.interfaces.annotations.NodeAccessType
import com.lessmarkup.interfaces.cache.DataCache
import com.lessmarkup.interfaces.structure.{NodeHandler, NodeHandlerFactory, ChildHandlerSettings}
import com.lessmarkup.userinterface.model.user.ForgotPasswordModel
import com.lessmarkup.userinterface.nodehandlers.common.DialogNodeHandler
import com.lessmarkup.{Constants, TextIds}

class ForgotPasswordNodeHandlerFactory @Inject() (dataCache: DataCache) extends NodeHandlerFactory {
  override def createNodeHandler(nodeHandlerConfiguration: NodeHandlerConfiguration, arguments: Any*): NodeHandler = {
    new ForgotPasswordNodeHandler(dataCache, nodeHandlerConfiguration)
  }
}

class ForgotPasswordNodeHandler (dataCache: DataCache, configuration: NodeHandlerConfiguration)
  extends DialogNodeHandler[ForgotPasswordModel](dataCache, classOf[ForgotPasswordModel], configuration) {

  protected override def getApplyCaption: String = {
    LanguageHelper.getText(Constants.ModuleTypeMain, TextIds.RESTORE_PASSWORD)
  }

  protected def loadObject: Option[ForgotPasswordModel] = {
    Option(DependencyResolver(classOf[ForgotPasswordModel]))
  }

  protected def saveObject(changedObject: Option[ForgotPasswordModel]): String = {
    if (changedObject.isDefined) {
      changedObject.get.submit(this, getFullPath)
    }
    null
  }

  override def createChildHandler(path: Seq[String]): Option[ChildHandlerSettings] = {

    if (path.length != 3 || path.head != "ticket") {
      return None
    }

    val nodeHandlerConfiguration = new NodeHandlerConfiguration(
      objectId = None,
      settings = None,
      accessType = NodeAccessType.READ,
      path = null,
      fullPath = null
    )

    val handler: ResetPasswordNodeHandler = createChildHandler(classOf[ResetPasswordNodeHandlerFactory], nodeHandlerConfiguration, path(1), path(2)).asInstanceOf

    Option(new ChildHandlerSettings(
      handler = handler,
      id = None,
      title = "",
      path = path.mkString("/"),
      rest = Nil
    ))
  }
}